src/renderer/helpers/copyAndPaste.ts

```diff
@@ -98,9 +102,37 @@ export const pasteFromClipboard = (
                 case 'text/rtf':
                     log.debug('Clipboard rtf', clipboard.readRTF());
                     break;
-                case 'image/png':
-                    log.debug('Clipboard image', clipboard.readImage().toPNG());
+                case 'image/jpeg':
+                case 'image/gif':
+                case 'image/bmp':
+                case 'image/tiff':
+                case 'image/png': {
+                    const tempDir = os.tmpdir();
+                    const imgData = clipboard.readImage().toPNG();
+                    const randName =
+                        (Math.random() + 1).toString(36).substring(7) +
+                        (Math.random() + 1).toString(36).substring(7) +
+                        (Math.random() + 1).toString(36).substring(7);
+                    const imgPath = path.join(tempDir, `chaiNNer-clipboard-${randName}.png`);
```

Comment:
Why not use a UUID as the random name?

```suggestion
                    const imgData = clipboard.readImage().toPNG();
                    const imgPath = path.join(os.tmpdir(), `chaiNNer-clipboard-${uuid4()}.png`);
```